posthog/temporal/tests/data_imports/test_end_to_end.py

```diff
@@ -2118,3 +2120,197 @@ async def test_append_only_table(team, mock_stripe_client):
     # We should now have 2 rows with the same `id`
     assert len(res.results) == 2
     assert res.results[0][0] == res.results[1][0]
+
+
+@pytest.mark.django_db(transaction=True)
+@pytest.mark.asyncio
+async def test_that_activity_triggers_schedule_buffer_one(team: Team, mock_stripe_client):
+    # Setup: create an external data source
+    source = await sync_to_async(ExternalDataSource.objects.create)(
+        source_id=uuid.uuid4(),
+        connection_id=uuid.uuid4(),
+        destination_id=uuid.uuid4(),
+        team=team,
+        status="running",
+        source_type="Stripe",
+        job_inputs={"stripe_secret_key": "test-key", "stripe_account_id": "acct_test"},
+    )
+
+    # Setup: create external data schema
+    schema = await sync_to_async(ExternalDataSchema.objects.create)(
+        name="BalanceTransaction",
+        team_id=team.pk,
+        source_id=source.pk,
+        sync_type=ExternalDataSchema.SyncType.INCREMENTAL,
+        sync_type_config={"incremental_field": "created", "incremental_field_type": "integer"},
+    )
+
+    # Create workflow input
+    inputs = ExternalDataWorkflowInputs(
+        team_id=team.pk,
+        external_data_source_id=source.pk,
+        external_data_schema_id=schema.pk,
+    )
+
+    # Test the workflow's error handling logic directly
+    # This simulates what happens when the workflow catches a WorkerShuttingDownError
+    with (
+        mock.patch(
+            "posthog.temporal.data_imports.external_data_job.a_trigger_schedule_buffer_one",
+            new_callable=mock.AsyncMock,
+        ) as mock_schedule,
+    ):
+        # Test the trigger_schedule_buffer_one_activity function directly
+        # This is what the workflow calls when it catches a WorkerShuttingDownError
+        from posthog.temporal.data_imports.external_data_job import trigger_schedule_buffer_one_activity
+
+        schedule_id = str(inputs.external_data_schema_id)
+        await trigger_schedule_buffer_one_activity(schedule_id)
+
+    # Ensure buffer-one trigger was called with the schema ID
+    mock_schedule.assert_awaited_once()
+    assert mock_schedule.call_args[0][1] == str(schema.pk)
```

Comment:
This test doesn't really belong here - the whole point of this file is to test the whole end-to-end functionality of the whole workflow. We can cover this test in an end-to-end fashion, though. I'd recommend removing this for the below